staging/src/k8s.io/client-go/openapi/groupversion.go

```diff
@@ -19,15 +19,16 @@ package openapi
 import (
 	"context"
 
-	openapi_v3 "github.com/google/gnostic/openapiv3"
-	"google.golang.org/protobuf/proto"
 	"k8s.io/kube-openapi/pkg/handler3"
 )
 
 const openAPIV3mimePb = "application/com.github.proto-openapi.spec.v3@v1.0+protobuf"
+const jsonMime = "application/json"
 
 type GroupVersion interface {
-	Schema() (*openapi_v3.Document, error)
+	// Raw data types
+	SchemaJSON() ([]byte, error)
+	SchemaPB() ([]byte, error)
```

Comment:
would it make more sense to have `Schema(contentType string) ([]byte, error)` and export the pb and json content types as constants? ideally we wouldn't have to revisit this interface for any additional content types in the future (and "SchemaPB" seems underspecified to always mean `"application/com.github.proto-openapi.spec.v3@v1.0+protobuf"`)

that's sort of similar to how the client patch methods take the content type as a string and the patch as `[]byte`

Reply:
Good points, I often like things to fail at compile time if we can, especially if the number of methods is limited, but I like  that your suggestion is consistent with how clients currently work in general.

Reply:
I've changed the interface to the style suggested by Jordan and updated tests for each content type